service/controller/resource_set.go

```diff
@@ -403,7 +403,7 @@ func NewResourceSet(config ResourceSetConfig) (*controller.ResourceSet, error) {
 			return false
 		}
 
-		if key.VersionBundleVersion(cr) == project.BundleVersion() {
+		if key.OperatorVersion(cr) == project.BundleVersion() {
```

Comment:
There is a separate task for this, but would it make sense to incorporate this change here as well:
https://github.com/giantswarm/giantswarm/issues/8637 ?

Reply:
Separate. No need to bother within this PR.